Include/cpython/code.h

```diff
@@ -88,6 +89,7 @@ typedef uint16_t _Py_CODEUNIT;
     PyObject *co_filename;        /* unicode (where it was loaded from) */     \
     PyObject *co_name;            /* unicode (name, for reference) */          \
     PyObject *co_qualname;        /* unicode (qualname, for reference) */      \
+    void* co_trampoline;                                               \
```

Comment:
Yeah, this is just for the prototype, I have dedicated no time whatsoever to make this maintainable

Reply:
+1, an internal prototype doesn't modify the interpreter so it uses `_PyCode_GetExtra` and `_PyCode_SetExtra` from logic to trampoline or not within a custom eval frame func via `_PyInterpreterState_SetEvalFrameFunc`.

The long term prognosis of this C stack trampoline approach is interesting in that we know python to python calls shouldn't need to use the C stack so having an eval frame hook at all could wind up as a feature on the chopping block for performance reasons.

It's also a little unfortunate to see trampolines consume more C stack space as that alters the applications stack consumption profile which works from a practicality beats purity standpoint but is not ideal.

Reply:
> The long term prognosis of this C stack trampoline approach is interesting in that we know python to python calls shouldn't need to use the C stack so having an eval frame hook at all could wind up as a feature on the chopping block for performance reasons.

We are already not using the inlining if a different python frame evaluator is being used. That needs to be preserved anyway and in any case, this feature could evolve in a way that if is an active python to python calls are not inlined. This is more than acceptable because perf is a sampling profiler and the trampoline is very thin, so it won't interfere with the samples because very little or none will end there. In any case, this is a common theme for profilers in general: the code you run needs to be introspectable in some ways so you never ran the same code at full speed.

Reply:
> Code objects really need fewer fields, not more.Would it be feasible to cram this into co_extra?

Done